### PR TITLE
Enable building latest megaservice image on push event in CI

### PR DESCRIPTION
**Pull request by daisy-ycguo**
## Description

Enable building latest megaservice image on push event in CI and pushing it to image registry

## Issues

n/a

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

n/a

## Tests

n/a

### CODE REVIEW

File: .github/workflows/image-build-on-push.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - "**/docker/ui/**"
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-on-push
+  cancel-in-progress: true
+
+jobs:
+  job1:
+    uses: ./.github/workflows/reuse-get-test-matrix.yml
```
Comment (chensuyue) [inline .github/workflows/image-build-on-push.yml:21]: Please verify this workflow on push.

Comment (daisy-ycguo) [reply .github/workflows/image-build-on-push.yml:21]: Yes, this is verified: https://github.com/opea-project/GenAIExamples/actions/runs/9515954706/job/26231251739
But I still need to verify if reuse-get-test-matrix works for pull_request_target.

Comment (daisy-ycguo) [reply .github/workflows/image-build-on-push.yml:21]: This has been verified in
- push: https://github.com/opea-project/GenAIExamples/actions/runs/9541582520/job/26294965584
- pull_request: https://github.com/opea-project/GenAIExamples/actions/runs/9541606106/job/26295017006
- pull_request_target: https://github.com/opea-project/GenAIExamples/actions/runs/9541606071/job/26295019580

**Review by chensuyue — APPROVED** [review]

**Review by XuehaoSun — APPROVED** [review]

**Comment by chensuyue** [comment]
Dockerfile path issue will be fixed in another PR.

**Pull request merged**